src/js/plugin.js

```diff
@@ -451,6 +451,26 @@ Plugin.BASE_PLUGIN_NAME = BASE_PLUGIN_NAME;
 
 Plugin.registerPlugin(BASE_PLUGIN_NAME, Plugin);
 
+/**
+ * Documented in player.js
+ *
+ * @ignore
+ */
+Player.prototype.usePlugin = function(name, plugin, ...args) {
```

Comment:
while this no longer registers it still keeps an inconsistent API.
You can't use this method on plugins that were already registered elsewhere and the way that plugins used via this method work is different than other plugins.